packages/react-dom/src/events/accumulateEnterLeaveListeners.js

```diff
@@ -83,24 +83,26 @@ function accumulateEnterLeaveListenersForEvent(
   }
   const dispatchListeners = [];
   const dispatchInstances = [];
+  const dispatchContainers = [];
 
   let node = target;
   while (node !== null) {
     if (node === common) {
       break;
     }
-    const alternate = node.alternate;
+    const {alternate, stateNode, tag} = node;
     if (alternate !== null && alternate === common) {
       break;
     }
-    if (node.tag === HostComponent) {
+    if (stateNode !== null && tag === HostComponent) {
```

Comment:
Is this a behavior change?

Either it was never null. And in that case it's safe to use it.

Or it was sometimes null. And then excluding this component from traversal changes the behavior without any flags.

Reply:
If we the stateNode is `null` it saves us processing the listener for the node as it's unmounted and we can avoid CPU cycles.

Reply:
What happens today if the node is unmounted in the middle of an event?

Reply:
If it were `null` today, we'd invoke `invokeGuardedCallbackAndCatchFirstError` with a `null` `currentTarget` on the synthetic event.

Reply:
If this is purely optimization I'd prefer to leave it out. Seems like unnecessary risk if this code is shared by the legacy system unless we have a clear proof that this does not change observable behavior.

Reply:
>we'd invoke invokeGuardedCallbackAndCatchFirstError with a null currentTarget on the synthetic event.

That seems like observably different behavior? User code would get called.

Reply:
Actually scrap what I said above, we call `getListener` later and that again checks for `stateNode` and if it's `null` then we don't fire any user code. So this change should be fine and have no obserable behavior changes.